presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -43,17 +43,27 @@ public PageBuilder(List<? extends Type> types)
 
     public PageBuilder(int initialExpectedEntries, List<? extends Type> types)
     {
+        this(initialExpectedEntries, DEFAULT_MAX_PAGE_SIZE_IN_BYTES, types);
+    }
+
+    public static PageBuilder withMaxPageSize(int maxPageBytes, List<? extends Type> types)
+    {
+        return new PageBuilder(Integer.MAX_VALUE, maxPageBytes, types);
+    }
+
+    private PageBuilder(int initialExpectedEntries, int maxPageBytes, List<? extends Type> types)
```

Comment:
The logic in this method seems very wrong to me. I know it's not particularly related to your change. But since you're adding a new kind of input here (setting initialExpectedEntries to INT_MAX), which would require thorough understanding of this constructor anyways. I would recommend rewriting this constructor.